### PR TITLE
Tenant ctrl: automatically delete namespace after timeout

### PR DESCRIPTION
**Pull request by giorio94**
# Description

This PR aggregates the modifications originally proposed in #821 (by @rs59) and #827 (by @QcFe), and preserved in the corresponding commits. Additionally, it introduces a set of minor improvements, as well as of dedicated tests.

Supersedes: #821, #827

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Automated tests
- [x] Staging environment

### CODE REVIEW

File: operators/pkg/tenant-controller/tenant_controller.go [hunk 1]
```diff
@@ -146,27 +149,26 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
...
 	}
 
+	// Determine if the personal namespace should be deleted
+
+	nsName := fmt.Sprintf("tenant-%s", strings.ReplaceAll(tn.Name, ".", "-"))
+
+	// Test if namespace has been open for too long; check if it is ok to delete
+	keepNsOpen, err := r.checkNamespaceKeepAlive(ctx, &tn, nsName)
+	if err != nil {
+		klog.Errorf("Error in r.List: unable to capture instances in tenant namespace %s -> %s", nsName, err)
```
Comment (QcFe) [inline operators/pkg/tenant-controller/tenant_controller.go:159]: Error message looks somewhat incoherent here

**Review by QcFe — APPROVED** [review]
lgtm! Just a minor comment

**Comment by giorio94** [comment]
@QcFe I've rebased this PR, solving the conflicts introduced by the new MyDrive feature and the removal of the old one. Before merging, please give it a try in the staging environment, to make sure everything still works properly.

**Comment by QcFe** [comment]
Thank you so much @giorio94, will go through tests on monday hopefully

**Comment by QcFe** [comment]
@giorio94 Sorry for the impossible delay. It seems there's a mismatch between the tenant cmd/main and the helm: the flag from helm is `tenant-ns-keep-alive` but the expected flag is `tenant-operator-ns-keep-alive`. I'll proceed with manual fix on the deployed staging to see everything else is correct

**Comment by giorio94** [comment]
> @giorio94 Sorry for the impossible delay. It seems there's a mismatch between the tenant cmd/main and the helm: the flag from helm is `tenant-ns-keep-alive` but the expected flag is `tenant-operator-ns-keep-alive`. I'll proceed with manual fix on the deployed staging to see everything else is correct

I've updated the PR to make the flags match. I've opted for `tenant-ns-keep-alive`, since this is related to tenants themselves, and not to the tenant operator.

**Pull request merged**